controller/controller_test.go

```diff
@@ -192,3 +194,104 @@ func Test_validateUnitStatusWithRequest(t *testing.T) {
 		}
 	}
 }
+
+func Test_matchesGroupSlices(t *testing.T) {
+	testCases := []struct {
+		InputUnitName string
+		InputRequest  Request
+		Output        bool
+	}{
+		{
+			"demo-main@1.service",
+			Request{
+				Group:    "demo",
+				SliceIDs: []string{"1", "2"},
+			},
+			true,
+		},
+		{
+			"demo-main@1.service",
+			Request{
+				Group:    "demo",
+				SliceIDs: []string{"3"},
+			},
+			false,
+		},
+
+		{
+			"demo-main.service",
+			Request{
+				Group:    "demo",
+				SliceIDs: nil,
+			},
+			true,
+		},
+
+		{
+			"other-main.service",
```

Comment:
I would like to see a test like this, but with `other-main@1.service`. This ensures that one matching criterium does not affect the overall result.